### PR TITLE
Modlog: prevent non-gstaff from searching for IPs

### PR DESCRIPTION
**Pull request by mia-pi-git**
Currently, if you /modlog [ip] (as a non-gstaff user) it will show actions with that IP,  even though the IP is removed. EG: https://prnt.sc/t10k1r
This prevents anyone who isn't gstaff from searching _just_ numbers, since without it it will include IP results.
I'm not sure if there's a better way to do this with regex but this blocks all cases i can think of and doesn't hurt functionality.

### CODE REVIEW

**Comment by Zarel** [comment]
Is this something we necessarily want to prevent?

File: server/chat-plugins/modlog.ts [hunk 1]
```diff
@@ -309,7 +309,10 @@ async function getModlog(
 	const addModlogLinks = !!(
 		Config.modloglink && (user.group !== ' ' || (targetRoom && targetRoom.settings.isPrivate !== true))
 	);
-
+	if (hideIps && /^[0-9.*]+$/.test(searchString)) {
```
Comment (Zarel) [inline server/chat-plugins/modlog.ts:312]:
```suggestion
	if (hideIps && /^\[?[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+\]?$/.test(searchString)) {
```

**Comment by mia-pi-git** [comment]
> Is this something we necessarily want to prevent?

I would say yes, as this rather defeats the point of hiding IPs.

**Comment by Zarel** [comment]
I disagree; "getting an IP" is different from "already having an IP and using it to get other information".

**Pull request merged**